server/src/main/java/fr/uga/l3miage/tp4/repositories/EcosSessionProgrammationStepRepository.java

```diff
@@ -0,0 +1,10 @@
+package fr.uga.l3miage.tp4.repositories;
+
+import fr.uga.l3miage.tp4.models.EcosSessionProgrammationStepEntity;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface EcosSessionProgrammationStepRepository extends JpaRepository<EcosSessionProgrammationStepEntity,Long> {
+        EcosSessionProgrammationStepEntity findTopByOrderByDateTimeDesc();
```

Comment:
la on ne sait pas sur quelle session , ici il fallait regarder la `session` sinon vous avez la dernière à faire mais sur toutes les sessions

```suggestion
        EcosSessionProgrammationStepEntity findTopByOrderByDateTimeDesc();
```

Reply:
et aussi vous n'êtes pas sur d'avoir un resultat donc `Optional< EcosSessionProgrammationStepEntity >`